YDotNet/Document/Types/Branches/Branch.cs

```diff
@@ -60,40 +63,47 @@ public IDisposable ObserveDeep(Action<IEnumerable<EventBranch>> action)
     ///     Starts a new read-write <see cref="Transaction" /> on this <see cref="Branch" /> instance.
     /// </summary>
     /// <returns>The <see cref="Transaction" /> to perform operations in the document.</returns>
-    /// <exception cref="YDotNetException">Another write transaction has been created and not commited yet.</exception>
+    /// <exception cref="YDotNetException">Another write transaction has been created and not committed yet.</exception>
     public Transaction WriteTransaction()
     {
-        ThrowIfDisposed();
-
-        var handle = BranchChannel.WriteTransaction(Handle);
-
-        if (handle == nint.Zero)
-        {
-            ThrowHelper.PendingTransaction();
-            return default!;
-        }
-
-        return new Transaction(handle, Doc);
+        return Doc.WriteTransaction();
     }
 
     /// <summary>
     ///     Starts a new read-only <see cref="Transaction" /> on this <see cref="Branch" /> instance.
     /// </summary>
     /// <returns>The <see cref="Transaction" /> to perform operations in the branch.</returns>
-    /// <exception cref="YDotNetException">Another write transaction has been created and not commited yet.</exception>
+    /// <exception cref="YDotNetException">Another write transaction has been created and not committed yet.</exception>
     public Transaction ReadTransaction()
     {
-        ThrowIfDisposed();
+        return Doc.ReadTransaction();
+    }
+
+    /// <summary>
+    ///     Gets a handle using the stored <see cref="BranchId" />.
+    /// </summary>
+    /// <param name="transaction">The transaction used to acquire the handle to the <see cref="Branch" />.</param>
+    /// <returns>The handle to the <see cref="Branch" />.</returns>
+    /// <exception cref="ObjectDisposedException">If <see cref="Branch.IsDisposed" /> is <c>true</c>.</exception>
+    protected internal nint GetHandle(Transaction transaction)
+    {
+        if (IsDisposed)
+        {
+            throw new ObjectDisposedException("Object is disposed.");
+        }
+
+        var handle = MemoryWriter.WriteStruct(BranchId);
```

Comment:
Using pattern would clarify the scope of the handle.

Reply:
In these cases I usually use the normal braces, because it is only needed for the next line and not the whole method, but it does not really matter.